homeassistant/components/reolink/__init__.py

```diff
@@ -84,6 +84,8 @@ async def async_device_config_update() -> None:
         async with asyncio.timeout(host.api.timeout * (RETRY_ATTEMPTS + 2)):
             try:
                 await host.update_states()
+            except CredentialsInvalidError as err:
+                raise ConfigEntryAuthFailed(err) from err
```

Comment:
This doesn't work as expected, as the data update coordinator will not trigger a re-auth flow when it encounters this exception.

Reply:
You could reload the integration, as it will then go into the normal flow.

Reply:
Are you sure?
I did not test this yet, but I looked at the dataupdatecoordinator code: 
https://github.com/home-assistant/core/blob/f85a8c504184b4787ec8d488814914ed42b0d794/homeassistant/helpers/update_coordinator.py#L361C1-L376C64

Under `except ConfigEntryAuthFailed` it has:
```
            if self.config_entry:
                self.config_entry.async_start_reauth(self.hass)
```

I presumed that would trigger a reauth...

Reply:
oh looks like you're right!